### PR TITLE
grunted changes

### PR DESCRIPTION
**Pull request by contactashish13**
#1003 

@EkoJR I have closed PR #2171 and created this one so that the changes do not confuse you. Can you please review this?

### CODE REVIEW

File: js/count-chars.js [hunk 1]
```diff
@@ -10,26 +10,36 @@
  * @version 2.9.2
  */
 
+ /* global aioseop_count_chars */
+
 var aiosp_title_extra = parseInt( aioseop_count_chars.aiosp_title_extra );
 
-jQuery( document ).ready( function() {
```
Comment (EkoJR) [inline js/count-chars.js]: All these jQuery functions are still being unnecessarily nested.

Comment (contactashish13) [reply js/count-chars.js]: I've resolved these .

File: js/quickedit_functions.js [hunk 2]
```diff
@@ -1,92 +1,96 @@
-jQuery(document).on( 'click', '.visibility-notice', function() {
+/* global aioseopadmin */
+/* global sack */
 
-    jQuery.ajax({
-        url: ajaxurl,
-        data: {
-            action: 'aioseo_dismiss_visibility_notice'
+jQuery( document ).on( 'click', '.visibility-notice', function() {
+    jQuery.ajax(
```
[the remark this answers is not in the document]
Comment (contactashish13) [reply js/quickedit_functions.js]: jquery.ajax functions are fine. I don't see why they should not be nested. The world expects them to be nested as they are different parameters.

File: js/modules/aioseop_opengraph.js [hunk 3]
```diff
@@ -9,13 +9,13 @@
...
-	}
-});
+jQuery( document ).ready( function () {
+    var snippet = jQuery( "#aioseop_snippet_link" );
+    if (snippet.length === 0) {
+        jQuery( "#aioseop_opengraph_settings_facebook_debug_wrapper" ).hide();
+    } else {
+        snippet = snippet.html();
+        jQuery( "#aioseop_opengraph_settings_facebook_debug" )
+        .attr( "href", "https://developers.facebook.com/tools/debug/sharing/?q=" + snippet );
```
[the remark this answers is not in the document]
Comment (contactashish13) [reply js/modules/aioseop_opengraph.js]: I think this fine. This issue is more about style than about minor/unimportant stuff such as this.

File: js/modules/aioseop_module.js [hunk 4]
```diff
@@ -207,391 +204,344 @@ function aioseop_add_condshow_handlers( index, value ) {
...
+    /**
+ * @since 1.0.0
+ * @return boolean.
+ */
+    jQuery(	"div#aiosp_video_sitemap_addl_pages_metabox" )
+    .delegate( "input[name='Submit']", "click", function() {
+        aioseop_handle_post_url(
+            'aioseop_ajax_save_url',
+            'video_sitemap_addl_pages',
+            jQuery( 'div#aiosp_video_sitemap_addl_pages_metabox input, div#aiosp_video_sitemap_addl_pages_metabox select' )
```
[the remark this answers is not in the document]
Comment (contactashish13) [reply js/modules/aioseop_module.js]: I feel this is minor/unimportant

File: js/modules/aioseop_module.js [hunk 5]
```diff
@@ -207,391 +204,344 @@ function aioseop_add_condshow_handlers( index, value ) {
...
-				);
-				return false;
-			}
-		);
+    /**
+ * @since 1.0.0
+ * @param $e.
+ * @return boolean.
+ */
+    jQuery( ".all-in-one-seo_page_all-in-one-seo-pack-modules-aioseop_feature_manager #aiosp_settings_form .aioseop_settings_left" )
```
[the remark this answers is not in the document]
Comment (contactashish13) [reply js/modules/aioseop_module.js]: there are many of this sort and this boils down to a change not a change of style. I think we should focus on style.

File: js/modules/aioseop_module.js [hunk 6]
```diff
@@ -187,16 +188,12 @@ function aioseop_do_condshow_match( index, value ) {
...
-				jQuery( '[name=' + subopt + ']' ).bind(
-					"change keyup", function() {
-						aioseop_do_condshow_match( index, value );
-					}
-				);
-			}
-		);
+	if ( typeof value !== 'undefined' ) {
+		jQuery.each( value, function(subopt, setting) {
+            jQuery( '[name=' + subopt + ']' ).bind( "change keyup", function() {
```
[the remark this answers is not in the document]
Comment (contactashish13) [reply js/modules/aioseop_module.js]: this is something left to a separate topic: different style guides indicated spaces or tabs. you can create a new issue for this. but here, lets focus on code correctness than style

File: js/modules/aioseop_opengraph.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (EkoJR) [reply js/modules/aioseop_opengraph.js]: It's a WP Standard for long lines like this. https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#chained-method-calls

> When a chain of method calls is too long to fit on one line, there must be one call per line, with the first call on a separate line from the object the methods are called on. If the method changes the context, an extra level of indentation must be used.

File: js/modules/aioseop_module.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (EkoJR) [reply js/modules/aioseop_module.js]: Was just a collaborative inquiry...

File: js/modules/aioseop_module.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (EkoJR) [reply js/modules/aioseop_module.js]: That seems concerning then.  Using tabs and not spaces for indentation is a WordPress Standard ( [PHP](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#spacing) & [JS](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#spacing) ), and I'm under the impression Grunt will run on every release, which would introduce these issues and break those WP standards.

It also appears that ESLint does provide settings for it. 
https://eslint.org/docs/2.0.0/rules/indent#rule-details
https://eslint.org/docs/4.0.0/rules/indent#options
https://eslint.org/docs/rules/no-mixed-spaces-and-tabs
https://eslint.org/docs/rules/no-tabs

It looked like a possible configuration could be...
```
indent: ['error', 'tab', {
	SwitchCase: 1,
}],
'no-mixed-spaces-and-tabs': 'warn',
'no-tabs': 'off',
```

[the remark this answers is not in the document]
Comment (EkoJR) [reply js/modules/aioseop_module.js]: Actually, it's the other way around...from what WP says.

> Your indentation should always reflect logical structure. Use real tabs and not spaces, as this allows the most flexibility across clients.

https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#indentation

>These rules encourage liberal spacing for improved developer readability. The minification process creates a file that is optimized for browsers to read and process.

> Indentation with tabs.

https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#spacing

There is a ESLint config provided by WPCS. Have you had a chance to check it out? https://github.com/WordPress-Coding-Standards/eslint-plugin-wordpress

[the remark this answers is not in the document]
Comment (contactashish13) [reply js/modules/aioseop_module.js]: If you use tabs and indent them to 4 and I use indented to 2 (depends on my liking and how many different environments I work under), it spoils readability so I feel this is part of a larger discussion of logical sanity vs. written rules. I prefer tabs, but spaces are more readable.

That is why I feel this is better suited to a more general discussion than put it as part of grunt. If ESLint needs to change, that forms a different issue. Grunt is an aggregator of processes and a beginning to some sort of semblance of code correctness that was previously absent. Configurations of individual processes might be improved at a later stage but that is more a WIP and should not hold back grunt.

**Comment by EkoJR** [comment]
I took a deeper look at grunt, and was able to get it running successfully on my computer. So, it seems something may have been fixed at some point.

I created a branch that has the same changes to grunt added by @contactashish13 , as well as re-enabled some of those rules and with an additional commit to the rules that fixed some of those problems I was having. When running grunt, the automatic changes were added, and I corrected what grunt logged for manual intervention. These were the changes made.

* [i-1003-1 branch commits](https://github.com/EkoJR/all-in-one-seo-pack/commits/i-1003-1).
* [Add changes from Grunt (phpcbf & phpcs)](https://github.com/EkoJR/all-in-one-seo-pack/commit/01ceff9d96f50bf6b3457949172e9300fe2b0f3c).
* [Add manual changes for Grunt errors/warnings (phpcs)](https://github.com/EkoJR/all-in-one-seo-pack/commit/aa306c7d0ed53916d403c85bf60b78a305093c93)
* [Manually grunted files (JSHint & ESLint)](https://github.com/EkoJR/all-in-one-seo-pack/commit/d8b15f6834a4205fcf9136aaac991c0ebd457653)

What seems odd, at no point did it ask to nest jQuery functions, nor did it replace the tabs with spaces. Those appeared to have been done manually and outside of grunt requirements; which is majority of the problems I had flagged. Perhaps something went wrong? ...or am I missing something?

Currently this PR would add more WPCS issues. With PHP, JS, & CSS, WordPress Standards says to use tabs, not spaces for indentation.

**Comment by contactashish13** [comment]
@EkoJR It looks like those changes were caused by my running phpcbf - for some reason it ran on the JS files as well instead of just targeting PHP.  We can discard my PR in favor of yours if it addresses all the problems you had highlighted. Glad you got it running on your computer!

**Comment by EkoJR** [comment]
Ok. We can use the other PR then. It has a commit to reflect your changes with grunt, as well as additional changes to get it operating on my end. Plus a few rules did get re-enabled since there was a way to ignore by line or code block, and still seemed like a useful rule; some of which were also marked `// TODO` to resolve in a later PR.

**Comment by EkoJR** [comment]
Closing this in favor of #2187

**Pull request closed without merging**